contrib/ledger-gen/src/programs.rs

```diff
@@ -0,0 +1,260 @@
+        send_and_confirm_transactions_in_parallel::{send_and_confirm_transactions_in_parallel_blocking, SendAndConfirmConfig}
+    },
+    solana_cli::{
+        program::calculate_max_chunk_size,
+    },
+    std::{
+        sync::Arc,
+    },
+};
+
+
+/// Sets Up Buffer Account in Order to Deploy Program
+///
+/// # Arguments
+/// * `client` - RPC Client
+/// * `payer` - Keypair of the Payer
+/// * `program_data` - Program Data
+///
+/// # Returns
+/// * `buffer_account` - Buffer Account Keypair
+///
+/// # Examples
+/// ```
+/// let buffer_account = programs::set_up_buffer_account_transactions(&client, &payer, &program_data);
+/// ```
+pub fn set_up_buffer_account_transactions(client: &Arc<RpcClient>, payer: &Keypair, program_data: &Vec<u8>) -> Keypair {
+    let blockhash = client.get_latest_blockhash().unwrap();
+    let buffer_account = Keypair::new();
+
+    let min_rent_exempt_program_data_balance = client.get_minimum_balance_for_rent_exemption(
+        UpgradeableLoaderState::size_of_programdata(program_data.len()),
+    ).unwrap();
+
+    let create_program_account_instruction = bpf_loader_upgradeable::create_buffer(
+        &payer.pubkey(),
+        &buffer_account.pubkey(),
+        &payer.pubkey(),
+        min_rent_exempt_program_data_balance,
+        program_data.len(),
+    );
+
+    let buffer_transaction = Transaction::new_signed_with_payer(
+        &create_program_account_instruction.unwrap(),
+        Some(&payer.pubkey()),
+        &[&payer, &buffer_account],
+        client.get_latest_blockhash().unwrap(),
+    );
+    let _ = client.send_and_confirm_transaction(&buffer_transaction).unwrap();
+
+    let create_msg = |offset: u32, bytes: Vec<u8>| {
+        let write_instruction = bpf_loader_upgradeable::write(
+            &buffer_account.pubkey(),
+            &payer.pubkey(),
+            offset,
+            bytes,
+        );
+        return Message::new_with_blockhash(&[write_instruction], Some(&payer.pubkey()), &blockhash);
+    };
+
+    let chunk_size = calculate_max_chunk_size(&create_msg);
+    // messages vector
+    let mut messages = vec![];
+    for (chunk, i) in program_data.chunks(chunk_size).zip(0..) {
+        let message = create_msg((i * chunk_size) as u32, chunk.to_vec());
+        messages.push(message);
+    }
+
+    let connection_cache = ConnectionCache::new_quic("connection_cache_cli_program_v4_quic", 1);
+
+    let cache = match connection_cache {
+        ConnectionCache::Quic(cache) => cache,
+        _ => unreachable!(),
+    };
+
+    let tpu_client_fut =
+        solana_client::nonblocking::tpu_client::TpuClient::new_with_connection_cache(
+            client.get_inner_client().clone(),
+            "ws://localhost:8900/",
+            solana_client::tpu_client::TpuClientConfig::default(),
+            cache,
+        );
+    let tpu_client = client
+        .runtime()
+        .block_on(tpu_client_fut)
+        .expect("Should return a valid tpu client");
+
```

Comment:
Would it make sense to have this as a seperate function/moved to utils? Seems like it could be handy in the future (and decomposing improves readability)

Reply:
dont want parallel transactions elsewhere